src/focus.service.ts

```diff
@@ -421,10 +380,17 @@ export class FocusService {
 
   public createArcEvent(direction: Direction): ArcEvent {
     const directional = isDirectional(direction);
+    let element: HTMLElement;
+    if (directional) {
+      element = this.findNextFocusByRaycast(direction);
+      if (!element) {
+        element = this.findNextFocus(direction);
```

Comment:
I do think that adding some lightweight metric insertion into the focus service would be beneficial though, both so that we can optimize layout so that fast paths such as this are hit more often, and so that we can tweak arcade-machine itself to be more performant.